# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmTypes.h
# Commit: 16fd9844f2d2
# Full Hash: 16fd9844f2d2029feaba3c864b1a0f830f9da235
# Author: Andy Wingo <wingo@igalia.com>
# Date: 2020-03-18 16:05:52
# Regressor Bug: 1620197
# File Overlap Count: 1
# Description:
#   Bug 1620197 - Enable multiple results from WebAssembly functions r=lth
#   
#   This patch enables multi-value calls and returns, adding some tests, and
#   conditionally disabling a couple expect-fail reftests that now pass.
#   
# ==============================================================================

diff -r 0d19968c99a7 -r 16fd9844f2d2 js/src/wasm/WasmTypes.h
--- a/js/src/wasm/WasmTypes.h	Wed Mar 18 13:39:50 2020 +0200
+++ b/js/src/wasm/WasmTypes.h	Wed Mar 18 10:29:53 2020 +0000
@@ -1086,6 +1086,16 @@
     }
     return false;
   }
+  // Entry from JS to wasm is currently unimplemented for functions that return
+  // multiple values.
+  bool temporarilyUnsupportedResultCountForEntry() const {
+    return results().length() > 1;
+  }
+  // Calls out from wasm to JS that return multiple values is currently
+  // unsupported.
+  bool temporarilyUnsupportedResultCountForExit() const {
+    return results().length() > 1;
+  }
   // For JS->wasm jit entries, AnyRef parameters and returns are allowed,
   // as are all reference types apart from TypeIndex.
   bool temporarilyUnsupportedReftypeForEntry() const {